terraform/state-persistence.tf

```diff
@@ -0,0 +1,9 @@
+data "terraform_remote_state" "gt-site-cluster" {
```

Comment:
Trying to read through some more docs and issues, it is still not entirely clear what this resource provides.

It does appear that remote state management has evolved a little. Apparently, after a `terraform remote config` invocation, `terraform` automatically pushes and pulls remote state.

I'm cool with leaving this is until we get to a point where I can take this PR for a spin, and then we'll be able to see how this remote state setup works on a machine that has no trace of existing Terraform state.